src/3d/qgs3dsceneexporter.h

```diff
@@ -0,0 +1,138 @@
+    int terrainResolution() const { return mTerrainResolution; }
+
+    //! Sets the terrain texture resolution
+    void setTerrainTextureResolution( int resolution ) { mTerrainTextureResolution = resolution; }
+    //! Returns the terrain resolution
+    int terrainTextureResolution() const { return mTerrainTextureResolution; }
+    //! Sets the scale of the exported 3D model
+    void setScale( float scale ) { mScale = scale; }
+    //! Returns the scale of the exported 3D model
+    float scale() const { return mScale; }
+
+  private:
+    //! Constructs Qgs3DExportObject from instanced point geometry
+    QVector<Qgs3DExportObject *> processInstancedPointGeometry( Qt3DCore::QEntity *entity, const QString &objectNamePrefix );
+    //! Constructs Qgs3DExportObject from 3D models loaded using a scene loader
+    QVector<Qgs3DExportObject *> processSceneLoaderGeometries( Qt3DRender::QSceneLoader *sceneLoader, const QString &objectNamePrefix );
+    //! Constructs Qgs3DExportObject from geometry renderer
+    Qgs3DExportObject *processGeometryRenderer( Qt3DRender::QGeometryRenderer *mesh, const QString &objectNamePrefix, float sceneScale = 1.0f, QVector3D sceneTranslation = QVector3D( 0.0f, 0.0f, 0.0f ) );
+    //! Extracts material information from geometry renderer and inserts it into the export object
+    void processEntityMaterial( Qt3DCore::QEntity *entity, Qgs3DExportObject *object );
+    //! Constricts Qgs3DExportObject from line entity
+    QVector<Qgs3DExportObject *> processLines( Qt3DCore::QEntity *entity, const QString &objectNamePrefix );
+    //! Constricts Qgs3DExportObject from billboard point entity
+    Qgs3DExportObject *processPoints( Qt3DCore::QEntity *entity, const QString &objectNamePrefix );
+
+    //! Returns a tile entity that contains the geometry to be exported and necessary scaling parameters
+    QgsTerrainTileEntity *getFlatTerrainEntity( QgsTerrainEntity *terrain, QgsChunkNode *node );
+    //! Returns a tile entity that contains the geometry to be exported and necessary scaling parameters
+    QgsTerrainTileEntity *getDemTerrainEntity( QgsTerrainEntity *terrain, QgsChunkNode *node );
+    //! Returns a tile entity that contains the geometry to be exported and necessary scaling parameters
+    QgsTerrainTileEntity *getMeshTerrainEntity( QgsTerrainEntity *terrain, QgsChunkNode *node );
+
+    //! Constructs a Qgs3DExportObject from the DEM tile entity
+    void parseDemTile( QgsTerrainTileEntity *tileEntity, const QString &layerName );
+    //! Constructs a Qgs3DExportObject from the flat tile entity
+    void parseFlatTile( QgsTerrainTileEntity *tileEntity, const QString &layerName );
+    //! Constructs a Qgs3DExportObject from the mesh terrain entity
+    void parseMeshTile( QgsTerrainTileEntity *meshEntity, const QString &layerName );
+
+    QString getObjectName( const QString &name );
+  private:
+    QMap<QString, int> usedObjectNamesCounter;
+    QVector<Qgs3DExportObject *> mObjects;
```

Comment:
I can't see where mObjects is cleaned up -- I believe the contents of this vector are leaked. Perhaps a destructor should be added which deletes all these objects?

Reply:
I made each object inside mObjects a child of QgsSceneExporter (that's why I made the class inherit QObject).
Now I changed that so that mObjects vector are deleted inside the destructor